addon/src/mocks/mock-request.js

```diff
@@ -139,55 +151,94 @@ export default class {
     return this;
   }
 
-  paramsMatch(request) {
+  // eslint-disable-next-line no-unused-vars
+  paramsMatch({ request, params }) {
+    const requestUrl = new URL(request.url, window.location.origin);
     if (!isEmptyObject(this.someQueryParams)) {
-      return isPartOf(
-        toParams(request.queryParams),
-        toParams(this.someQueryParams),
-      );
+      // request must have every specified item in someQueryParams, but can have more.
+      const someQueryParams = new URLSearchParams(param(this.someQueryParams)); // convert json -> URLSearchParams
+      return someQueryParams
+        .keys()
+        .every((key) =>
+          isEqual(
+            someQueryParams.getAll(key),
+            requestUrl.searchParams.getAll(key),
+          ),
+        );
```

Comment:
largely the same, just trying to use more native javascript to do the heavy lifting. In this case `URL` and `URLSearchParams` provide us a bunch for free